starbucks_dw/macros/calculate_response_rate.sql

```diff
@@ -0,0 +1,10 @@
+{% macro calculate_response_rate(column_name, statuses, total_count_column, alias) %}
```

Comment:
I think you are overthinking it by using a for loop. This is my suggestion:

```suggestion
{% macro calculate_response_rate(transaction_type_col, transaction_type, total_response, column_name) %}
    count(*) filter (where {{ transaction_type_col }} = '{{ transaction_type }}')
    / count({{ total_response }}) as {{ column_name }}
{% endmacro %}
```